tests/unit/s2n_rsa_pss_rsae_test.c

```diff
@@ -285,12 +293,20 @@ int main(int argc, char **argv)
                 &rsa_cert_chain->cert_chain->head->raw));
         EXPECT_EQUAL(rsa_pkey_type, S2N_PKEY_TYPE_RSA);
 
-        RSA *rsa_key = EVP_PKEY_get0_RSA(rsa_public_key.pkey);
+        /* In openssl 3 the function `EVP_PKEY_get1_RSA` returns a cached copy of PKEY
+         * rather than the original copy. Therefore to modify the value of
+         * rsa_public_key, `EVP_PKEY_set1_RSA` must be called and rsa_key_copy
+         * must be freed.
+         *
+         * Modify the rsa_public_key for the new test_case */
+        RSA *rsa_key_copy = EVP_PKEY_get1_RSA(rsa_public_key.pkey);
```

Comment:
I dunno if I like the name rsa_key_copy. Maybe "mutable_key" because we can actually modify it?

Reply:
In ossl3 we get a pre-cached copy of the key. So `mutable` would be misleading. Its better to think of this as a `copy` of the key, which we can modify, doesnt change the underlying key and we must free after use.

Reply:
Honestly I just have no idea what "pre-cached copy of a key" means 😆. Also your comment is incorrect, in this part of the test we aren't altering an rsa_pss_public_key value.

I think the comment here should be:
```
/* Initialize the rsa_public_key for each test_case */
```
I don't think going into the details of how the key gets initialized will help anyone trying to understand the test.

Reply:
> I just have no idea what "pre-cached copy of a key" means 

- [evp_pkey_st](https://github.com/openssl/openssl/blob/709d4be78f64a8ba0707fb5682b90039e848dad4/include/crypto/evp.h#L666-L670) (== EVP_PKEY) contains `legacy_pkey_st pkey` and `legacy_pkey_st legacy_cache_pkey`
- [`EVP_PKEY_getx_RSA` calls `evp_pkey_get_legacy`](https://github.com/openssl/openssl/blob/31b7f23d2f958491d46c8a8e61c2b77b1b546f3e/crypto/evp/p_legacy.c#L43-L55)
- which returns [`legacy_cache_pkey.ptr`](https://github.com/openssl/openssl/blob/af801ec89205aaf6ebf8522d510d0b1fc29e3233/crypto/evp/p_lib.c#L2093)